### DISCUSSION TITLE
Storybook-Tailwindcss integration in Angular v16 project

### DISCUSSION THREAD

**Discussion by rlovasz** [storybookjs/storybook#22716] [Help]
Hi everyone,

I am having an issue to make storybook and tailwind work together in an Angular 16 app.

This is my repo: https://github.com/rlovasz/tailwind-storybook
I have followed the tailwind installation guide: https://tailwindcss.com/docs/guides/angular
And then the storybook tailwind integration guide (written for react apps): https://storybook.js.org/recipes/tailwindcss

This is on my console:
`npm run storybook

> tailwind-storybook@0.0.0 storybook
> ng run tailwind-storybook:storybook

info => Starting manager..
WARN No story files found for the specified pattern: src/**/*.mdx
(node:24317) ExperimentalWarning: The Fetch API is an experimental feature. This feature could change at any time
(Use `node --trace-warnings ...` to show where the warning was created)
info Addon-docs: using MDX2
info => Using implicit CSS loaders
info => Using angular browser target options from "tailwind-storybook:build"
info => Using angular project with "tsConfig:/home/rlovasz/Development/tailwind-storybook/.storybook/tsconfig.json"
info => Using default Webpack5 setup
<i> [webpack-dev-middleware] wait until bundle finished
99% end closing watch compilationWARN Force closed preview build
ModuleParseError: Module parse failed: Unexpected character '@' (1:0)
File was processed with these loaders:
 * ./node_modules/resolve-url-loader/index.js
 * ./node_modules/sass-loader/dist/cjs.js
You may need an additional loader to handle the result of these loaders.
> @tailwind base;
| @tailwind components;
| @tailwind utilities;
    at handleParseError (/home/rlovasz/Development/tailwind-storybook/node_modules/webpack/lib/NormalModule.js:742:23)
    at /home/rlovasz/Development/tailwind-storybook/node_modules/webpack/lib/NormalModule.js:844:9
    at processResult (/home/rlovasz/Development/tailwind-storybook/node_modules/webpack/lib/NormalModule.js:598:14)
    at /home/rlovasz/Development/tailwind-storybook/node_modules/webpack/lib/NormalModule.js:647:9
    at /home/rlovasz/Development/tailwind-storybook/node_modules/loader-runner/lib/LoaderRunner.js:368:5
    at iterateNormalLoaders (/home/rlovasz/Development/tailwind-storybook/node_modules/loader-runner/lib/LoaderRunner.js:213:12)
    at iterateNormalLoaders (/home/rlovasz/Development/tailwind-storybook/node_modules/loader-runner/lib/LoaderRunner.js:217:12)
    at /home/rlovasz/Development/tailwind-storybook/node_modules/loader-runner/lib/LoaderRunner.js:229:5
    at context.callback (/home/rlovasz/Development/tailwind-storybook/node_modules/loader-runner/lib/LoaderRunner.js:116:16)
    at onSuccess (/home/rlovasz/Development/tailwind-storybook/node_modules/resolve-url-loader/index.js:168:9)

Broken build, fix the error above.
You may need to refresh the browser.
`

I assume that this is probably not a bug, just the react guide is not working for angular. Could anyone help?

**Comment by micobarac** [comment]
I am having hard time integrating Angular 17 and Tailwind 3.3.6 with Storybook 7.6.3. I am using scss style file format. How do I get Tailwind CSS to process `*.stories.ts` files? I tried setting `content` in `tailwind.config.js` to `./src/**/*.{html,ts}`, but Tailwind CSS classes are not applied to stories.

**Reply by Marklb** [reply]
I followed the Tailwind steps for installing in an Angular project, made sure the styles worked if I ran `ng serve`.  Then I added added Storybook and started it, without doing any other than adding a story for a component that is using Tailwind classes, and the Tailwind styles seem to be in my story.

Storybook is using Angular's builder, so if the styles work when you run `ng serve` then they should work in Storybook. If you are adding any custom styling rules then you may be overwriting Angular's loaders.

**Comment by Marklb** [comment]
I don't know much about Tailwind, but I took a look and it seems to work, without anything special needing to be added.

@rlovasz I don't know if you still need an answer, but in your reproduction I removed `@storybook/addon-styling` from `.storybook/main.ts` and I removed `import '../src/styles.scss';` from `.storybook/preview.ts` and the Tailwinds styles seem to load fine.
